### PR TITLE
[ci] Fix several release tests

### PR DESCRIPTION
**Pull request by can-anyscale**
Fix several release tests. I need to send out instructions on how to write tests with byod soon so people avoid breaking it.

- tune_cloud_long_running_cloud_storage has to run on python 3.8
- agent_stress_test needs some additional ml packages
- rllib tests

Testing Strategy
   - [x] Release tests

### CODE REVIEW

**Comment by can-anyscale** [comment]
Test runs:
- https://buildkite.com/ray-project/release-tests-pr/builds/44303
- https://buildkite.com/ray-project/release-tests-pr/builds/44305

File: release/release_tests.yaml [hunk 1]
```diff
@@ -3676,7 +3676,7 @@
     script: pytest checkpointing_tests/test_learner_group_checkpointing.py
 
     wait_for_nodes:
-      num_nodes: 3
+      num_nodes: 2
```
Comment (aslonnie) [inline release/release_tests.yaml:3679]: why is this node number change required?

**Review by aslonnie — APPROVED** [review]

**Comment by aslonnie** [comment]
https://buildkite.com/ray-project/release-tests-pr/builds/44305 is still failing?

File: release/release_tests.yaml (hunk 1 above)
Comment (can-anyscale) [reply release/release_tests.yaml:3679]: A previous PR reduces the number of node to 2 https://github.com/ray-project/ray/pull/36929, so this needs to update as well

**Comment by can-anyscale** [comment]
Some tests still fail but they are the second layer of failures after my fix. I'll leave them for rllib folks to debug them further then.

**Review by avnishn — APPROVED** [review]

**Review by matthewdeng — APPROVED** [review]
<div><br/>via <a href="https://giphy.com/happyplaceshow/">Happy Place</a> on <a href="https://giphy.com/gifs/happyplaceshow-tcE1V02xBAOngb2agQ">GIPHY</a></div>

**Pull request merged**